File: src/main/kotlin/com/andy/ik/util/IOUtil.java
package com.andy.ik.util;

import java.io.Closeable;


public class IOUtil {

    public static void closeQuietly(Closeable... closeables) {
        for (Closeable closeable : closeables) {
            closeQuietly(closeable);
        }
    }

    public static void closeQuietly(Closeable closeable) {
        if (closeable != null) {
            try {
                closeable.close();
            } catch (RuntimeException rethrown) {
                throw rethrown;
            } catch (Exception ignored) {
            }
        }
    }
}

File: ikt/src/main/kotlin/com/andy/ikt/util/IOUtil.kt
package com.andy.ikt.util

import java.io.Closeable


object IOUtil {

    fun closeQuietly(vararg closeables: Closeable) {
        closeables.map { closeQuietly(it) }
    }

    fun closeQuietly(closeable: Closeable?) {
        closeable?.apply {
            try {
                close()
            } catch (rethrown: RuntimeException) {
                throw rethrown
            } catch (ignored: Exception) {
            }
        }
    }
}
